src/scalar.jl

```diff
@@ -173,21 +311,21 @@ as(::Type{Real}, left, right) =
 
 as(::Type{Real}, ::Infinity{false}, ::Infinity{true}) = Identity()
 
-as(::Type{Real}, left::Real, ::Infinity{true}) = ShiftedExp(true, left)
```

Comment:
Is there any usecase remaining for `ShiftedExp` etc.? Shouldn't these be removed in the PR?

Reply:
Removal is fine with me.